File: say.js
const fetch = require('node-fetch');

// Function to convert text to speech
async function text_to_speech(text) {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/audio/text_to_speech";
    const payload = {"providers": "google,amazon", "language": "en-US", "option": "MALE", "text": text};
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(payload)
    });
    const result = await response.json();
    return result['google']['audio'];
}

// Function to convert text to speech asynchronously
async function text_to_speech_async(text) {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/audio/text_to_speech_async";
    const payload = {"providers": "lovoai", "language": "en-US", "option": "MALE", "text": text};
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(payload)
    });
    const result = await response.json();
    return result;
}

// Function for automatic translation
async function automatic_translation(text, source_language, target_language) {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/translation/automatic_translation";
    const payload = {"providers": "google,amazon", 'source_language': source_language, 'target_language': target_language, 'text': text};
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(payload)
    });
    const result = await response.json();
    return result['google']['text'];
}

// Function for speech to text conversion
async function speech_to_text_async(file_url) {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/audio/speech_to_text_async";
    const json_payload = {"providers": "google, amazon", "language": "en-US", "file_url": file_url};
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(json_payload)
    });
    const result = await response.json();
    return result;
}

// Function for text generation
async function generate_text() {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/text/generation";
    const payload = {
        "providers": "openai,cohere",
        "text": "this is a test of Eden AI",
        "temperature": 0.2,
        "max_tokens": 250
    };
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(payload)
    });
    const result = await response.json();
    return result['openai']['generated_text'];
}

// Function for spell check
async function spell_check(text) {
    const headers = {"Authorization": "Bearer 🔑 Your_API_Key"};
    const url = "https://api.edenai.run/v2/text/spell_check";
    const payload = {"providers": "openai,microsoft", "language": "en", "text": text};
    const response = await fetch(url, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(payload)
    });
    const result = await response.json();
    return result['openai']['items'];
}

// Example usage
const text = "Hello, how are you?";
text_to_speech(text)
    .then(audio => console.log(audio));

const translated_text = automatic_translation(text, "en", "fr");
console.log(translated_text);

const file_url = "🔗 URL of your audio file";
speech_to_text_async(file_url)
    .then(transcribed_text => console.log(transcribed_text));

generate_text()
    .then(generated_text => console.log(generated_text));

spell_check(text)
    .then(spell_checked_text => console.log(spell_checked_text));

File: say.cpp
#include <iostream>
#include <string>
#include <curl/curl.h>
#include <jsoncpp/json/json.h>

// Function to convert text to speech
std::string text_to_speech(std::string text) {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/audio/text_to_speech";
    std::string providers = "google,amazon";
    std::string language = "en-US";
    std::string option = "MALE";

    CURL *curl;
    CURLcode res;
    std::string response;

    curl_global_init(CURL_GLOBAL_DEFAULT);
    curl = curl_easy_init();
    if (curl) {
        struct curl_slist *headers = NULL;
        headers = curl_slist_append(headers, ("Authorization: Bearer " + api_key).c_str());
        headers = curl_slist_append(headers, "Content-Type: application/json");

        Json::Value payload;
        payload["providers"] = providers;
        payload["language"] = language;
        payload["option"] = option;
        payload["text"] = text;

        Json::StreamWriterBuilder writer;
        std::string json_payload = Json::writeString(writer, payload);

        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        curl_easy_setopt(curl, CURLOPT_POSTFIELDS, json_payload.c_str());
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response);

        res = curl_easy_perform(curl);

        curl_slist_free_all(headers);
        curl_easy_cleanup(curl);
    }

    curl_global_cleanup();

    Json::Value result;
    Json::CharReaderBuilder reader;
    std::istringstream json_response(response);
    std::string err;
    Json::parseFromStream(reader, json_response, &result, &err);

    return result["google"]["audio"].asString();
}

// Function to convert text to speech asynchronously
std::string text_to_speech_async(std::string text) {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/audio/text_to_speech_async";
    std::string providers = "lovoai";
    std::string language = "en-US";
    std::string option = "MALE";

    // TODO: Implement the function

    return "";
}

// Function for automatic translation
std::string automatic_translation(std::string text, std::string source_language, std::string target_language) {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/translation/automatic_translation";
    std::string providers = "google,amazon";

    // TODO: Implement the function

    return "";
}

// Function for speech to text conversion
std::string speech_to_text_async(std::string file_url) {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/audio/speech_to_text_async";
    std::string providers = "google,amazon";
    std::string language = "en-US";

    // TODO: Implement the function

    return "";
}

// Function for text generation
std::string generate_text() {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/text/generation";
    std::string providers = "openai,cohere";
    std::string text = "this is a test of Eden AI";
    double temperature = 0.2;
    int max_tokens = 250;

    // TODO: Implement the function

    return "";
}

// Function for spell check
std::string spell_check(std::string text) {
    std::string api_key = "🔑 Your_API_Key";
    std::string url = "https://api.edenai.run/v2/text/spell_check";
    std::string providers = "openai,microsoft";
    std::string language = "en";

    // TODO: Implement the function

    return "";
}

int main() {
    std::string text = "Hello, how are you?";
    std::string audio = text_to_speech(text);
    std::cout << audio << std::endl;

    std::string translated_text = automatic_translation(text, "en", "fr");
    std::cout << translated_text << std::endl;

    std::string file_url = "🔗 URL of your audio file";
    std::string transcribed_text = speech_to_text_async(file_url);
    std::cout << transcribed_text << std::endl;

    std::string generated_text = generate_text();
    std::cout << generated_text << std::endl;

    std::string spell_checked_text = spell_check(text);
    std::cout << spell_checked_text << std::endl;

    return 0;
}
